### PR TITLE
Add scaling support of embedded page to ui-iframe node

### PR DESCRIPTION
**Pull request by HiroyasuNishiyama**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

This PR add  scaling feature of embedded web page to `ui-iframe` node.
An example of the new feature can be found in `Import`/`Examples`/`node-red-node-ui-iframe`/`04 - Scale embedded web page.json`.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: node-red-node-ui-iframe/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "node-red-node-ui-iframe",
-    "version": "0.1.2",
+    "version": "0.1.3",
```
Comment (dceejay) [inline node-red-node-ui-iframe/package.json:3]: Should bump to 0.2.0 as it's a new feature

File: node-red-node-ui-iframe/ui_iframe.html [hunk 2]
```diff
@@ -30,8 +30,9 @@
         <input type="text" id="node-input-url">
     </div>
     <div class="form-row">
-        <label for="node-input-origin"><i class="fa fa-external-link"></i> <span data-i18n="ui_iframe.label.origin"></span></label>
```
Comment (dceejay) [inline node-red-node-ui-iframe/ui_iframe.html:33]: Did you mean to remove this option ? (Indeed not sure it it is useful but just checking)

File: node-red-node-ui-iframe/ui_iframe.js [hunk 3]
```diff
@@ -44,6 +61,9 @@ module.exports = function(RED) {
            if (msg.url) {
                iframe.setAttribute("src", msg.url);
            }
+           if (msg.scale) {
+               setScale(iframe, msg.scale);
```
Comment (dceejay) [inline node-red-node-ui-iframe/ui_iframe.js:65]: Do you need to bounds check this for valid values ?

File: node-red-node-ui-iframe/ui_iframe.html [hunk 4]
```diff
@@ -78,6 +80,14 @@
                     height: "#node-input-height",
                     group: "#node-input-group"
                 });
+                $("#node-input-scale").spinner({
+                    min:1,
+                    max:100
```
Comment (dceejay) [inline node-red-node-ui-iframe/ui_iframe.html:85]: so 100% (full size) is max ? IE you can't enlarge as well as reduce ?

**Review by dceejay — COMMENTED** [review]
Few little things for feedback

File: node-red-node-ui-iframe/package.json (hunk 1 above)
Comment (HiroyasuNishiyama) [reply node-red-node-ui-iframe/package.json]: Bumped version to 0.2.0

File: node-red-node-ui-iframe/ui_iframe.html (hunk 2 above)
Comment (HiroyasuNishiyama) [reply node-red-node-ui-iframe/ui_iframe.html:33]: Sorry something wrong with code merge. reverted the code.

File: node-red-node-ui-iframe/ui_iframe.js (hunk 3 above)
Comment (HiroyasuNishiyama) [reply node-red-node-ui-iframe/ui_iframe.js:67]: Thank you. Yes. Need validation.
Added the check in `setScale` function.

File: node-red-node-ui-iframe/ui_iframe.html (hunk 4 above)
Comment (HiroyasuNishiyama) [reply node-red-node-ui-iframe/ui_iframe.html]: It can be any positive value.  
Removed the limits.

Comment (dceejay) [reply node-red-node-ui-iframe/ui_iframe.html]: I think limits would be useful - but think 1 is too small and 100 may not be enough... would 20 - 200 make sense ?

File: node-red-node-ui-iframe/ui_iframe.js [hunk 5]
```diff
@@ -22,17 +22,36 @@ module.exports = function(RED) {
...
-<div style="width:100%; height:100%; display:inline-block;">
+<div id="${id}-div" style="width:100%; height:100%; display:inline-block;">
     <iframe id="${id}" src="${url}" allow="${allow}" style="width:100%; height:100%; overflow:hidden; border:0; display:block">
         Failed to load Web page
     </iframe>
 </div>
 <script>
 (function(scope) {
+    function setScale(iframe, scale) {
+        if (scale > 0) {
```
Comment (dceejay) [inline node-red-node-ui-iframe/ui_iframe.js:39]: should it check for being a number also ? (or at least be parseable into a number)

File: node-red-node-ui-iframe/ui_iframe.html (hunk 4 above)
Comment (HiroyasuNishiyama) [reply node-red-node-ui-iframe/ui_iframe.html]: Sounds good.  Added limits.

File: node-red-node-ui-iframe/ui_iframe.js (hunk 5 above)
Comment (HiroyasuNishiyama) [reply node-red-node-ui-iframe/ui_iframe.js]: I think only possible value except number that passes this condition is `true` value.  So added check on this.
Also added error reporting.

**Comment by dceejay** [comment]
Published to npm - many thanks

**Pull request merged**